types/request.d.ts

```diff
@@ -30,6 +30,7 @@ export interface FastifyRequest<
   hostname: string;
   url: string;
   method: string;
+  validationError: Record<string, object>; // in order for this to be used the user should ensure they have set the attachValidation option.
```

Comment:
I think this is not correct definition for `validationError`.

According to current implementation validationError can be `undefined` or `Error` with additional properties - `validation` which contains actual errors and `validationContext` - part where's validation failed (body, headers etc).

cc @Ethan-Arrowood, @mcollina  correct me if I'm wrong :)

```suggestion
  validationError: undefined | Error & { validation: any; validationContext: string }; // in order for this to be used the user should ensure they have set the attachValidation option.
```